src/app/COMPONENTS/task-details/task-details.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core'
+import { Task } from '../../models/task.model'
+import { MaterialModule } from '../../material.module'
+import { TasksService } from '../../service/tasks.service'
+import { ActivatedRoute } from '@angular/router'
+import { Observable } from 'rxjs'
+import { CommonModule, NgIf } from '@angular/common'
+import { FormBuilder, FormControl, Validators, ReactiveFormsModule, FormGroup } from '@angular/forms'
+
+@Component({
+  selector: 'app-task-details',
+  standalone: true,
+  imports: [
+    MaterialModule,
+    CommonModule,
+    NgIf,
+    ReactiveFormsModule
+  ],
+  templateUrl: './task-details.component.html',
+  styleUrl: './task-details.component.scss'
+})
+export class TaskDetailsComponent implements OnInit {
+  ngOnInit() {
+    if (this.activatedRoute.snapshot.paramMap) {
+      this.taskId = this.activatedRoute.snapshot.paramMap.get('id')
+    }
+    if (this.taskId) {
+      this.task$ = this.tasksService.getTaskById(this.taskId)
+    }
+  }
+  constructor(private tasksService: TasksService, private activatedRoute: ActivatedRoute, private fb: FormBuilder) {
+
+    this.taskForm = this.fb.group({
+      title: new FormControl({ value: "", disabled: true }, Validators.required),
+      description: new FormControl({ value: "", disabled: true }, Validators.required),
```

Comment:
define variables at the start of the component as mentioned in the previous comment